pandas/tests/test_common.py

```diff
@@ -251,8 +256,53 @@ def test_compression_size_fh(obj, method, compression_only):
         assert uncompressed > compressed
 
 
-# GH 21227
+@pytest.mark.parametrize('write_method, write_kwargs, read_method', [
+    ('to_csv', {'index': False}, pandas.read_csv),
+    ('to_json', {}, pandas.read_json),
+    ('to_pickle', {}, pandas.read_pickle),
+])
+def test_dataframe_compression_defaults_to_infer(
+        write_method, write_kwargs, read_method, compression_only):
+    # Test that DataFrame.to_* methods default to inferring compression from
+    # paths. https://github.com/pandas-dev/pandas/pull/22011
+    input = DataFrame([[1.0, 0, -4.4], [3.4, 5, 2.4]], columns=['X', 'Y', 'Z'])
+    extension = _compression_to_extension[compression_only]
+    with tm.ensure_clean('compressed' + extension) as path:
+        # assumes that compression='infer' is the default
```

Comment:
If you can reduce the amount of comments that would be ideal. Citing this as a specific example it doesn't add anything that's not already known from the test name. 

Generally we just go `# GHXXXXX` as a comment at the beginning of the test to reference the original issue